Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM python:3.10.8 as base
+RUN pip install poetry
+ENV WEBAPP_PORT=80
+EXPOSE ${WEBAPP_PORT}
+COPY . /app
+WORKDIR /app
+RUN poetry install
```

Comment:
You're currently running `poetry install` after copying all app files. This works, but has the unintended consequence that, if one of the files in the codebase changes, when you rebuild the image you'll need to re-install all poetry dependencies. This happens because the docker layer generated by line 5 gets invalidated, so the other layers are no longer cached.

A good way of optimizing it would be to:
* `WORKDIR /app`
* `COPY poetry.lock pyproject.toml /app`
* `RUN poetry install`
* `COPY . /app`

Now, you're installing the packages before copying the app files, so if a file changes in the codebase, only the layers starting from `COPY . ./app` are invalidated, and all previous ones (including poetry install) are cached. This should save quite some time in subsequent builds.